src/tests/ftest/util/yaml_utils.py

```diff
@@ -0,0 +1,469 @@
+                    # test_server/test_client arguments
+                    quantity = len(replacement)
+                elif self._override:
+                    self.log.warning(
+                        "Warning: In order to override the node quantity a "
+                        "'--test_clients' argument must be specified: %s: %s",
+                        key, placeholder)
+                for _ in range(quantity):
+                    try:
+                        replacement_nodes.add(replacement.pop(0))
+                    except IndexError as error:
+                        # Not enough nodes provided for the replacement
+                        message = f"Not enough '{key}' placeholder replacements specified"
+                        self.log.error("  - %s; required: %s", message, quantity)
+                        raise YamlException(message) from error
+
+            except ValueError:
+                try:
+                    # Replace clush-style placeholders with nodes from the user provided list using
+                    # a mapping so that values used more than once yield the same replacement
+                    for node in NodeSet(placeholder):
+                        if node not in node_mapping:
+                            try:
+                                node_mapping[node] = replacement.pop(0)
+                            except IndexError as error:
+                                # Not enough nodes provided for the replacement
+                                if not self._override:
+                                    message = f"Not enough '{key}' placeholder replacements remain"
+                                    self.log.error(
+                                        "  - %s; required: %s", message, NodeSet(placeholder))
+                                    raise YamlException(message) from error
+                                break
+                            self.log.debug(
+                                "  - %s replacement node mapping: %s -> %s",
+                                key, node, node_mapping[node])
+                        replacement_nodes.add(node_mapping[node])
+
+                except TypeError as error:
+                    # Unsupported format
+                    message = f"Unsupported placeholder format: {placeholder}"
+                    self.log.error(message)
+                    raise YamlException(message) from error
+
+            hosts_key = r":\s+".join([key, str(placeholder)])
+            hosts_key = hosts_key.replace("[", r"\[")
+            hosts_key = hosts_key.replace("]", r"\]")
+            if replacement_nodes:
+                replacements[hosts_key] = ": ".join([key, str(replacement_nodes)])
+            else:
+                replacements[hosts_key] = None
+
+    @staticmethod
+    def _get_storage_replacement(replacements, placeholder_data, key, replacement):
+        """_summary_.
```

Comment:
Do we need summary and descriptions for this method?

Reply:
Yes. Fixed, thanks.